### PR TITLE
CP-15401: Show/Hide rdpPolling, upsell and optimize for XenApp depend…

### PR DESCRIPTION
**Pull request by cheng--zhang**
…ing on XenCenter registry keys

Below feature will  made available depending on registry keys are:

VM Advanced options -> Optimize for Citrix XenApp
Checkbox "Enable Remote Desktop console scanning (XenServer 6.5 and earlier)" from Options dialog (Console page)
Links in the Upsell dialogs (this refers to "Learn More" buttons on the Upsell dialogs/Tab pages)

Signed-off-by: Cheng Zhang <cheng.zhang@citrix.com>

### CODE REVIEW

**Comment by cheng--zhang** [comment]
Hi @MihaelaStoica 
I already update the reg key as we talked.
Could you please help to review the code and give your comments?
Thanks.

File: XenAdmin/Alerts/Types/MessageAlert.cs [hunk 1]
```diff
@@ -348,8 +348,10 @@ public override Action FixLinkAction
 
 					case XenAPI.Message.MessageType.LICENSE_EXPIRES_SOON:
 					case XenAPI.Message.MessageType.LICENSE_DOES_NOT_SUPPORT_POOLING:
-						return () => Program.OpenURL(InvisibleMessages.LICENSE_EXPIRY_WEBPAGE);
-
+                        if (XenAdmin.Core.Registry.LinkLabelHidden)
+                            return null;
+                        else
+                            return () => Program.OpenURL(XenAdmin.Core.Registry.LinkLabelHidden ? "" : InvisibleMessages.LICENSE_EXPIRY_WEBPAGE);
```
Comment (MihaelaStoica) [inline XenAdmin/Alerts/Types/MessageAlert.cs]: LinkLabelHidden check shouldn't be needed here, as it's already tested above.

File: XenAdmin/Diagnostics/Problems/ConnectionProblem/VmprActivatedWarning.cs [hunk 2]
```diff
@@ -55,7 +55,9 @@ public override string Description
 
         public override Uri UriToLaunch
         {
-            get { return new Uri(InvisibleMessages.VMPR_DEPRECATION_URL); }
+            get {
```
Comment (MihaelaStoica) [inline XenAdmin/Diagnostics/Problems/ConnectionProblem/VmprActivatedWarning.cs]: I think that the Uri(string) constructor will raise a UriFormatException if the string is empty.
We cound instead do:
return XenAdmin.Core.Registry.LinkLabelHidden ? null : new Uri(InvisibleMessages.VMPR_DEPRECATION_URL);
Also classes WarningWithInformationUrl and ProblemWithInformationUrl need to cope with UriToLaunch==null.

File: XenAdmin/Diagnostics/Problems/ConnectionProblem/VmprActivatedWarning.cs [hunk 3]
```diff
@@ -55,7 +55,9 @@ public override string Description
 
         public override Uri UriToLaunch
         {
-            get { return new Uri(InvisibleMessages.VMPR_DEPRECATION_URL); }
+            get {
+                return new Uri(XenAdmin.Core.Registry.LinkLabelHidden ? "" : InvisibleMessages.VMPR_DEPRECATION_URL);
```
Comment (MihaelaStoica) [inline XenAdmin/Diagnostics/Problems/ConnectionProblem/VmprActivatedWarning.cs]: I think that the Uri(string) constructor will raise a UriFormatException if the string is empty.
We cound instead do:
return XenAdmin.Core.Registry.LinkLabelHidden ? null : new Uri(InvisibleMessages.VMPR_DEPRECATION_URL);
Also classes WarningWithInformationUrl and ProblemWithInformationUrl need to cope with UriToLaunch==null.

**Comment by cheng--zhang** [comment]
Hi @MihaelaStoica 
I have reworked the code by your suggestion in the new commit. 
Could you please help to review the code again?
Thanks a lot.

**Comment by cheng--zhang** [comment]
Hi @MihaelaStoica 
I have complete the rework of comments.
Please help to review the latest commit and merge if possible.
Thanks a lot.

**Comment by MihaelaStoica** [comment]
Looks good. Merging.

**Pull request merged**